x/gov/client/cli/prompt_test.go

```diff
@@ -33,11 +40,49 @@ func TestPromptIntegerOverflow(t *testing.T) {
 	for _, intOverflower := range intOverflowers {
 		overflowStr := intOverflower
 		t.Run(overflowStr, func(t *testing.T) {
-			readline.Stdout.Write([]byte(overflowStr + "\n"))
+			origStdin := readline.Stdin
+			defer func() {
+				readline.Stdin = origStdin
+			}()
+
+			fin, fw := readline.NewFillableStdin(os.Stdin)
+			readline.Stdin = fin
+			fw.Write([]byte(overflowStr + "\n"))
 
 			v, err := cli.Prompt(st{}, "")
-			assert.NotNil(t, err, "expected a report of an overflow")
 			assert.Equal(t, st{}, v, "expected a value of zero")
+			require.NotNil(t, err, "expected a report of an overflow")
+			require.Contains(t, err.Error(), "range")
+		})
+	}
+}
+
+func TestPromptParseInteger(t *testing.T) {
+	// Intentionally sending a value out of the range of
+	values := []struct {
+		in   string
+		want int
+	}{
+		{fmt.Sprintf("%d", math.MinInt), math.MinInt},
+		{"19991", 19991},
+		{"991000000199", 991000000199},
+	}
+
+	for _, tc := range values {
+		tc := tc
+		t.Run(tc.in, func(t *testing.T) {
+			origStdin := readline.Stdin
+			defer func() {
+				readline.Stdin = origStdin
+			}()
+
+			fin, fw := readline.NewFillableStdin(os.Stdin)
+			readline.Stdin = fin
+			fw.Write([]byte(tc.in + "\n"))
+
+			v, err := cli.Prompt(st{}, "")
+			assert.Nil(t, err, "expected a nil error")
+			assert.Equal(t, tc.want, v.I, "expected a value of zero")
```

Comment:
```suggestion
			assert.Equal(t, tc.want, v.I, "expected equal values")
```

Reply:
I used assert because I we want don't want just 1 line to fail, we want both reports. I initially had require but it obstructed reaching the other lines :-)